src/pages/vaults/VaultDepositWithdrawForm.tsx

```diff
@@ -138,16 +146,81 @@ export const VaultDepositWithdrawForm = ({
     dispatch(setVaultFormConfirmationStep(true));
   };
 
-  const onSubmitConfirmForm = () => {
-    // TODO tell abacus and respond
-    onSuccess?.();
+  const { depositToMegavault, withdrawFromMegavault } = useSubaccount();
+  const forceRefreshVaultAccount = useForceRefreshVaultAccount();
+  const forceRefreshVault = useForceRefreshVaultDetails();
+  const notify = useCustomNotification();
+
+  const onSubmitConfirmForm = async () => {
+    if (isSubmitting) {
+      return;
+    }
+    setIsSubmitting(true);
+    try {
+      const { submissionData } = validationResponse;
+      if (operation === 'DEPOSIT') {
+        const cachedAmount = submissionData?.deposit?.amount;
+        if (cachedAmount == null) {
+          notify({
+            title: 'Unable to submit Megavault Deposit transaction',
```

Comment:
[q] guessing we're going to add l10n strings later right? (i'm 100% fine with that, just want to check)

Reply:
yeah all notifications set to be rewritten